backend/runner/runner.go

```diff
@@ -420,17 +424,20 @@ func (s *Service) registrationLoop(ctx context.Context, send func(request *ftlv1
 	})
 	if err != nil {
 		s.registrationFailure.Store(optional.Some(err))
+		observability.Runner.RegistrationFailure(ctx, deploymentKey, state)
 		return fmt.Errorf("failed to register with Controller: %w", err)
 	}
 	s.registrationFailure.Store(optional.None[error]())
 
 	// Wait for the next heartbeat.
 	delay := s.config.HeartbeatPeriod + time.Duration(rand.Intn(int(s.config.HeartbeatJitter))) //nolint:gosec
 	logger.Tracef("Registered with Controller, next heartbeat in %s", delay)
+	observability.Runner.Registered(ctx, deploymentKey, state)
 	select {
 	case <-ctx.Done():
 		err = context.Cause(ctx)
 		s.registrationFailure.Store(optional.Some(err))
+		observability.Runner.RegistrationFailure(ctx, deploymentKey, state)
```

Comment:
aside: seeing these make me wonder whether we should be pushing up the actual errors to DD as well

Reply:
Yeah that has come up a couple times. TBH I'm not sure either. Maybe we should bring this up with the larger team and @bradleydwyer since he has quite a bit of experience here as well.

Reply:
+1, looking forward to hearing what everyone has to say